### PR TITLE
feat(detection): several enhancement and reformat

### PR DESCRIPTION
**Pull request by wjfwzzc**
1. change the definition of the parameter of smooth L1 loss to support L1 loss
2. change the default regression loss to L1 loss
3. load pretrained weights from hubs
4. re-organize directories
5. reformat codes

Weights are being trained and will be uploaded soon.

### CODE REVIEW

File: official/vision/detection/models/retinanet.py [hunk 1]
```diff
@@ -31,13 +31,15 @@ def __init__(self, cfg, batch_size):
             anchor_scales=self.cfg.anchor_scales,
             anchor_ratios=self.cfg.anchor_ratios,
         )
-        self.box_coder = layers.BoxCoder(reg_mean=cfg.reg_mean, reg_std=cfg.reg_std)
+        self.box_coder = layers.BoxCoder(cfg.reg_mean, cfg.reg_std)
 
         self.stride_list = np.array([8, 16, 32, 64, 128]).astype(np.float32)
```
Comment (FateScript) [inline official/vision/detection/models/retinanet.py:36]: 我觉得这两行进config吧

File: official/vision/detection/tools/data_mapper.py [hunk 2]
```diff
@@ -8,8 +8,4 @@
 # "AS IS" BASIS, WITHOUT ARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 from megengine.data.dataset import COCO, Objects365, PascalVOC
 
-data_mapper = dict(
-    coco=COCO,
-    objects365=Objects365,
-    voc=PascalVOC,
-)
+data_mapper = dict(coco=COCO, objects365=Objects365, voc=PascalVOC,)
```
Comment (FateScript) [inline official/vision/detection/tools/data_mapper.py:11]: 分三行写呗，后面好加

Comment (wjfwzzc) [reply official/vision/detection/tools/data_mapper.py:11]: done

File: official/vision/detection/layers/det/box_utils.py [hunk 3]
```diff
@@ -82,17 +88,13 @@ def encode(self, bbox: Tensor, gt: Tensor) -> Tensor:
         target_dh = F.log(gt_height / bbox_height)
         target = self._concat_new_axis(target_dx, target_dy, target_dw, target_dh)
 
-        if self.reg_mean is not None:
-            target -= self.reg_mean
-        if self.reg_std is not None:
-            target /= self.reg_std
+        target -= self.reg_mean
```
[the remark this answers is not in the document]
Comment (wjfwzzc) [reply official/vision/detection/layers/det/box_utils.py:91]: 没必要省这个。

File: official/vision/detection/README.md [hunk 4]
```diff
@@ -90,18 +89,16 @@ nvcc -I $MGE/_internal/include -shared -o lib_nms.so -Xcompiler "-fno-strict-ali
 在得到训练完保存的模型之后，可以通过tools下的test.py文件测试模型在`COCO2017`验证集的性能：
 
 ```bash
-python3 tools/test.py -f retinanet_res50_coco_1x_800size.py \
-                      -n 8 \
-                      --model /path/to/retinanet_weights.pt \
-                      --dataset_dir /data/datasets
+python3 tools/test.py -f configs/retinanet_res50_coco_1x_800size.py -n 8 \
+                      -m /path/to/retinanet_weights.pt \
```
[the remark this answers is not in the document]
Comment (wjfwzzc) [reply official/vision/detection/README.md:93]: done

File: official/vision/detection/models/retinanet.py (hunk 1 above)
Comment (wjfwzzc) [reply official/vision/detection/models/retinanet.py:36]: done

File: README.md [hunk 5]
```diff
@@ -73,12 +73,12 @@ export PYTHONPATH=/path/to/models:$PYTHONPATH
...
+目标检测同样是计算机视觉中的常见任务，我们提供了两个经典的目标检测模型[Retinanet](./official/vision/detection/model/retinanet)和[Faster R-CNN](./official/vision/detection/model/faster_rcnn)，这两个模型在**COCO验证集**上的测试结果如下：
 
-| 模型                              | mAP<br>@5-95    |
-| :---:                             | :---:           |
-| retinanet-res50-1x-800size        | 36.0            |
-| faster-rcnn-fpn-res50-1x-800size  | 37.3            |
+| 模型                         | mAP<br>@5-95 |
+| :---:                        | :---:        |
+| retinanet-res50-1x-800size   | 36.4         |
+| faster-rcnn-res50-1x-800size | 38.8         |
```
[the remark this answers is not in the document]
Comment (wjfwzzc) [reply README.md:81]: 看成名字就好了，没必要特别标注。

**Comment by FateScript** [comment]
LGTM

**Pull request merged**